docs/guides/server/hostname2.adoc

```diff
@@ -0,0 +1,161 @@
+
+[source,bash]
+----
+kc.sh start --hostname https://my.keycloak.org
+----
+
+Similarly, your reverse proxy might expose Keycloak at a different context path. It is possible to configure Keycloak to reflect that via the `hostname` and `hostname-admin` options. See the following example:
+
+[source,bash]
+----
+kc.sh start --hostname https://my.keycloak.org:123/auth
+----
+
+== Utilizing an internal URL for communication among clients
+
+Keycloak has the capability to offer a separate URL for backchannel requests, enabling internal communication while maintaining the use of a public URL for frontchannel requests. Moreover, the backchannel is dynamically resolved based on incoming headers. Consider the following example:
+
+[source,bash]
+----
+kc.sh start --hostname https://my.keycloak.org --hostname-backchannel-dynamic true
+----
+
+In this manner, your applications, referred to as clients, can connect with {project_name} through your local network, while the server remains publicly accessible at https://my.keycloak.org[https://my.keycloak.org].
+
+== Using edge TLS termination
+
+As you can observe, the HTTPS protocol is the default choice, adhering to {project_name}'s commitment to security best practices. However, {project_name} also provides the flexibility for users to opt for HTTP if necessary. This can be achieved simply by specifying the HTTP listener, consult the https://www.keycloak.org/server/enabletls[TLS guide] for details. With an edge TLS-termination proxy you can start the server as follows:
+
+[source,bash]
+----
+kc.sh start --hostname https://my.keycloak.org --http-enabled true
+----
+
+The result of this configuration is that you can continue to access {project_name} at https://my.keycloak.org[https://my.keycloak.org] via HTTPS, while the proxy interacts with the instance using HTTP and port `8080`.
+
+== Relying on a trusted proxy for the dynamic hostname resolution
+
+If you fully trust your proxy, which correctly sets the Forwarded header, and you don't want to hardcode the hostname, Keycloak can accommodate this. You simply need to initiate the server as follows:
+
+[source,bash]
+----
+kc.sh start --hostname-strict false --proxy-headers forwarded
+----
+
+With this configuration, the server respects the value set by the Forwarded header. This also implies that all endpoints are dynamically resolved.
+
+The `--proxy-headers` option can be also used to resolve the URL partially dynamically when `--hostname` option is not specified as a full URL. For example:
+
+[source,bash]
+----
+kc.sh start --hostname my.keycloak.org --proxy-headers forwarded
+----
+
+In this case, scheme, port and context path are resolved dynamically from X-Forwarded-* headers, while hostname is statically defined as my.keycloak.org[my.keycloak.org].
```

Comment:
I'd put some security note at the end of this section. Something like:
https://github.com/keycloak/keycloak/blob/811c70d1361c60dfd52abc7932a16715cd00a952/docs/guides/server/reverseproxy.adoc?plain=1#L25

Reply:
@Pepo48 I believe this is still valid.

Reply:
Yes, it is, it slipped through as I was applying the suggestions through GH UI. Sorry for that, it is included now.